src/BaseStyles.tsx

```diff
@@ -26,6 +26,10 @@ const GlobalStyle = createGlobalStyle<{colorScheme?: 'light' | 'dark'}>`
   details-dialog:focus:not(:focus-visible):not(.focus-visible) {
     outline: none;
   }
+
+  p a[class*=StyledLink] {
```

Comment:
Is there a way you could test this in some of our React apps? I like the idea of it, but I feel like this could easily cause unexpected visual changes. Underlined in a body of text probably won't ruffle any feathers, but there's probably some cases where its not a body of text and more like one line of text as part of a UI.

Reply:
I agree with @langermank , but also have no idea where to start. @colebemis - maybe you can help?

Reply:
I think if this is a blocker then we could wait until we receive more guidance on `Link` and underline usage across Primer. I know this is the case for the PVC `Link` accessibility fixes, so if this is required here too then I think it's okay to come back to this.

If we wanted to test this we'd probably have to query up the usage of PRC `Link` across GitHub and find the instances where this component is a descendant of a paragraph element, though I would love to know if there are better ways to go about this. From a very quick look, this seems like an uncommon pairing, so there might not be too much friction, but that's only from a small sample size.

Reply:
Agree with @TylerJDev that this may be better solved more holistically through upcoming work to determine underline styles for links in https://github.com/github/primer/issues/2099.

Reply:
Rolled back the changes around styling links in text. This way, we can merge in the `as` restriction, and let the more holistic work handle this issue.